document-store/src/integrationTest/java/org/hypertrace/core/documentstore/postgres/PostgresDocStoreTest.java

```diff
@@ -108,6 +112,59 @@ public void testUpsertAndReturn() throws IOException {
     Assertions.assertEquals(document.toJson(), resultDocument.toJson());
   }
 
+  @Test
+  public void test_getJsonNodeAtPath() throws Exception {
+    ObjectMapper objectMapper = new ObjectMapper();
+    ObjectNode labelNode = objectMapper.createObjectNode();
+    labelNode.put("string", "Label2");
+
+    ObjectNode valueNode = objectMapper.createObjectNode();
+    valueNode.set("value", labelNode);
+
+    ArrayNode valuesNode = objectMapper.createArrayNode();
+    valuesNode.add(valueNode);
+
+    ObjectNode valueListNode = objectMapper.createObjectNode();
+    valueListNode.set("values", valuesNode);
+
+    ObjectNode labelsNode = objectMapper.createObjectNode();
+    labelsNode.set("valueList", valueListNode);
+
+    ObjectNode attributesNode = objectMapper.createObjectNode();
+    attributesNode.put("name", "testKey2");
+    attributesNode.set("labels", labelsNode);
+
+    ObjectNode rootNode = objectMapper.createObjectNode();
+    rootNode.put("id", "testKey2");
+    rootNode.set("attributes", attributesNode);
+    rootNode.put("created_at", "2022-07-12 17:46:03.750437");
+    rootNode.put("updated_at", "2022-07-12 17:46:03.750437");
+
+    String path1 = "attributes.labels.valueList.values";
+    String outputNode1 = valuesNode.toString();
+    JsonNode expectedRootNode1 = objectMapper.readTree(outputNode1);
+
+    String path2 = "attributes.labels.attrNotPresent.values";
+    String outputNode2 = "[]";
+    JsonNode expectedRootNode2 = objectMapper.readTree(outputNode2);
+    PostgresCollection collection = (PostgresCollection) datastore.getCollection(COLLECTION_NAME);
+    try {
+      Assertions.assertEquals(
+          collection.getJsonNodeAtPath(path1, rootNode, true), expectedRootNode1);
+      Assertions.assertEquals(
+          collection.getJsonNodeAtPath(path2, rootNode, true), expectedRootNode2);
+      Assertions.assertEquals(collection.getJsonNodeAtPath(null, rootNode, true), rootNode);
+      Assertions.assertEquals(
+          collection.getJsonNodeAtPath(path1, rootNode, false), expectedRootNode1);
+      Assertions.assertEquals(
```

Comment:
Q: If, we would have called this method first before creating if not present, what would be the output? It would be an exception, right?
```
Assertions.assertEquals(
          collection.getJsonNodeAtPath(path2, rootNode, false), expectedRootNode2)
```

Reply:
I think, I am referring to this, it would be this, right?
```
Assertions.assertEquals(collection.getJsonNodeAtPath(null, rootNode, false), rootNode);
    } catch (Exception e) {
      System.out.println("Created path is not right");
      Assertions.fail();
    }
  }
```

Reply:
if path is null then method will return rootNode. createPathIfMissing value of this flag will not matter in that case.

Reply:
Earlier it was an exception..you are right..I have added a check to handle the same